x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -46,3 +58,9 @@ fn get_value(program: Program, of: Address) -> i64 {
         .get(StateKeys::Counter(of).to_vec())
         .expect("failed to get counter")
 }
+
+/// Gets the count at the address for an external program.
+#[public]
+fn get_value_external(program: Program, target: Program, max_units: i64, of: Address) -> i64 {
+    program.call_program(&target, max_units, "get_value", &[Box::new(of)])
+}
```

Comment:
I thought of a way that this can be done directly in the `#[public]` macro. The `inc` and `get_value` can be compiled behind a `cfg(lib)` and `cfg(not(lib))` where you can pull another program in as a dependency. 

The limitation here would be that you couldn't design a program that calls another deployment of itself, but I'm not sure that's a bad thing.

Reply:
sounds interesting please create an issue for that so we can track it.

Reply:
#523 for tracking